transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java

```diff
@@ -535,6 +536,28 @@ public void testLastHandlerEmptyPipeline() throws Exception {
         assertNull(pipeline.last());
     }
 
+    @Test(timeout = 5000)
+    public void testChannelInitializerException() throws Exception {
+        final AtomicReference<Throwable> error = new AtomicReference<Throwable>();
+        final CountDownLatch latch = new CountDownLatch(1);
+        EmbeddedChannel channel = new EmbeddedChannel(new ChannelInitializer<Channel>() {
+            @Override
+            protected void initChannel(Channel ch) throws Exception {
+                throw new IllegalStateException();
```

Comment:
Instantiate this exception outside the channel scope so you can do `assertSame` at the end of the test?